### PR TITLE
Add some fixes to the script

### PR DESCRIPTION
**Pull request by hadrich-hatem**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.3.x
| Description?  | This PR adds some fixes to the script.
| Type?         | improvement
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | /
| How to test?  | /

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9006)

### CODE REVIEW

File: tests/E2E/test/campaigns/common_scenarios/module.js [hunk 1]
```diff
@@ -8,7 +9,16 @@ module.exports = {
   installModule: function (client, ModulePage, AddProductPage, moduleTechName) {
     test('should go to "Module" page', () => client.goToSubtabMenuPage(Menu.Improve.Modules.modules_menu, Menu.Improve.Modules.modules_services_submenu));
     test('should set the name of the module in the search input', () => client.waitAndSetValue(ModulePage.module_selection_input, moduleTechName));
-    test('should click on "Search" button', () => client.waitForExistAndClick(ModulePage.selection_search_button));
+    test('should click on "Search" button', () => {
+      return promise
+        .then(() => client.waitForExistAndClick(ModulePage.selection_search_button))
+        .then(() => client.isVisible(AddProductPage.symfony_toolbar, 3000))
```
Comment (mickaelandrieu) [inline tests/E2E/test/campaigns/common_scenarios/module.js]: You can disable the debug mode by overriding the configuration, isn't a better solution than click on Symfony debug toolbar to hide it?

Comment (mickaelandrieu) [reply tests/E2E/test/campaigns/common_scenarios/module.js]: You may create `defines_custom.inc.php` file in `config` folder and set `_PS_MODE_DEV_` to false /c @mbadrani

Comment (mickaelandrieu) [reply tests/E2E/test/campaigns/common_scenarios/module.js]: ping @fatmaBouchekoua wdyt about my idea? it's better than click on debug toolbar to hide it on every e2e test, isn't it?

Comment (PierreRambaud) [reply tests/E2E/test/campaigns/common_scenarios/module.js]: I'm really agree with you, I think the Symfony debug toolbar must be disable while running e2e tests.

Comment (fatmaBouchekoua) [reply tests/E2E/test/campaigns/common_scenarios/module.js]: These tests can be launched on local, and if we will disable the debug mode, developers will not have the choice to run them with the debug mode enabled.
So I think that we have to hide the Symfony debug toolbar if it's visible in some cases (only if that toolbar is hiding a button)

Comment (mickaelandrieu) [reply tests/E2E/test/campaigns/common_scenarios/module.js]: Every user/developer can disable debug mode, you may also add a check in e2e to stop the test suite and inform the developer that the test suite is only available for a shop in production mode.

Not only you don't need to rely on DOM to disable the debug toolbar but the test suite will run faster as there is a really big difference of performance between dev and production mode (~150% on some benchmarks).

**Review by mickaelandrieu — DISMISSED** [review]

File: tests/E2E/test/campaigns/common_scenarios/module.js [hunk 2]
```diff
@@ -8,7 +9,11 @@ module.exports = {
   installModule: function (client, ModulePage, AddProductPage, moduleTechName) {
     test('should go to "Module" page', () => client.goToSubtabMenuPage(Menu.Improve.Modules.modules_menu, Menu.Improve.Modules.modules_services_submenu));
     test('should set the name of the module in the search input', () => client.waitAndSetValue(ModulePage.module_selection_input, moduleTechName));
-    test('should click on "Search" button', () => client.waitForExistAndClick(ModulePage.selection_search_button));
+    test('should click on "Search" button', () => {
+      if (global.ps_mode_dev)
+        promise = client.waitForExistAndClick(AddProductPage.symfony_toolbar);
```
Comment (PierreRambaud) [inline tests/E2E/test/campaigns/common_scenarios/module.js:14]: can write:

```js
return client.waitForExistAndClick(AddProductPage.symfony_toolbar).then(() => client.waitForExistAndClick(ModulePage.selection_search_button));
```

File: tests/E2E/test/campaigns/common_scenarios/product.js [hunk 3]
```diff
@@ -95,17 +95,12 @@ module.exports = {
...
-            .then(() => client.isVisible(AddProductPage.symfony_toolbar))
-            .then(() => {
-              if (global.isVisible) {
-                client.waitForExistAndClick(AddProductPage.symfony_toolbar)
-              }
-            })
-            .then(() => client.waitForExistAndClick(AddProductPage.save_product_button))
+        test('should switch the product online', () => {
+          if (global.ps_mode_dev)
+            promise = client.waitForExistAndClick(AddProductPage.symfony_toolbar);
```
Comment (PierreRambaud) [inline tests/E2E/test/campaigns/common_scenarios/product.js:100]: Can write return on a single line

File: tests/E2E/test/campaigns/full/14_discount/3_check_discount_in_FO.js [hunk 4]
```diff
@@ -69,6 +70,21 @@ scenario('Create "Catalog price rule"', () => {
...
+    test('should search for the created product', () => client.searchProductByName(productData["name"] + date_time));
+    test('should click on "Edit" button', () => client.waitForExistAndClick(ProductList.edit_button));
+    test('should click on "Pricing" tab', () => client.waitForExistAndClick(AddProductPage.product_pricing_tab));
+    test('should click on "Delete" button of specific price', () => client.scrollWaitForExistAndClick(AddProductPage.specific_price_delete_button, 150));
+    test('should click on "Yes" of modal button', () => client.waitForVisibleAndClick(AddProductPage.continue_confirmation));
+    test('should verify the appearance of the green validation', () => client.checkTextValue(AddProductPage.validation_msg, 'Successful deletion'));
+    test('should click on "Save" button', () => {
+      if (global.ps_mode_dev)
+        promise = client.waitForExistAndClick(AddProductPage.symfony_toolbar);
+      return promise.then(() => client.waitForExistAndClick(AddProductPage.save_product_button, 3000));
```
Comment (PierreRambaud) [inline tests/E2E/test/campaigns/full/14_discount/3_check_discount_in_FO.js:84]: Same here

File: tests/E2E/test/campaigns/common_scenarios/module.js (hunk 2 above)
Comment (fatmaBouchekoua) [reply tests/E2E/test/campaigns/common_scenarios/module.js:14]: Hello @PierreRambaud ,
We can't click on the Symfony debug toolbar if the debug mode is disabled

**Pull request closed without merging**